unittest/mysys/byte_order-t.c

```diff
@@ -0,0 +1,100 @@
+/* Copyright (c) 2019, MariaDB
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+/**
+  @file
+
+  Unit tests for serialization and deserialization functions
+*/
+
+#include "tap.h"
+
+#include "my_byteorder.h"
+#include "myisampack.h"
+
+void test_byte_order()
+{
+  MY_ALIGNED(CPU_LEVEL1_DCACHE_LINESIZE)
+  uchar buf[CPU_LEVEL1_DCACHE_LINESIZE * 2];
+
+  uchar *aligned= buf;
+  uchar *not_aligned= buf + CPU_LEVEL1_DCACHE_LINESIZE - 1;
+
+#define TEST(STORE_NAME, LOAD_NAME, TYPE, VALUE, BYTES)                        \
+  {                                                                            \
+    TYPE value= VALUE;                                                         \
+    uchar bytes[]= BYTES;                                                      \
+    STORE_NAME(aligned, value);                                                \
+    ok(!memcmp(aligned, bytes, sizeof(bytes)), "");                            \
+    ok(LOAD_NAME(aligned) == value, "");                                       \
+    STORE_NAME(not_aligned, value);                                            \
+    ok(!memcmp(not_aligned, bytes, sizeof(bytes)), "");                        \
+    ok(LOAD_NAME(not_aligned) == value, "");                                   \
+  }
+
+#define ARRAY_2(A, B) {A, B}
+#define ARRAY_3(A, B, C) {A, B, C}
+#define ARRAY_4(A, B, C, D) {A, B, C, D}
+#define ARRAY_5(A, B, C, D, E) {A, B, C, D, E}
+#define ARRAY_6(A, B, C, D, E, F) {A, B, C, D, E, F}
+#define ARRAY_7(A, B, C, D, E, F, G) {A, B, C, D, E, F, G}
+#define ARRAY_8(A, B, C, D, E, F, G, H) {A, B, C, D, E, F, G, H}
```

Comment:
Why all these defines? Can't you feed `{ A, B }` directly?

Reply:
I can't put it directly because of a comma. `#define COMMA ,` is uglier.

Reply:
Can't we have just one buffer like `{ 0xFE, 0xDC, 0xBA, 0x98, 0x76, 0x54, 0x32, 0x10 }` and use required amount of left most bytes. For negative value testing use one more reverse buffer?

Reply:
I've thought about it. And didn't invent how to do it without ugliness.